File: sdk/ios/Sources/DuallerSDK/NavigationHandler.swift
import UIKit

/// Handles page navigation
public final class NavigationHandler: Handler {
    public let apiName: String = "dualler.navigateTo"

    public func handle(_ message: BridgeMessage) -> String? {
        guard let payload = message.payload,
              let url = payload["url"] as? String else {
            return "{\"errMsg\":\"invalid navigation URL\"}"
        }
        // In production, would present a new ViewController
        print("[Dualler] Navigate to: \(url)")
        return "{}"
    }
}

File: sdk/android/src/main/java/com/dualler/sdk/handler/NavigationHandler.kt
package com.dualler.sdk.handler

import android.app.Activity
import com.dualler.sdk.protocol.APIContract
import com.dualler.sdk.protocol.BridgeMessage

class NavigationHandler(private val activity: Activity) : Handler {
    override val apiName: String = APIContract.NAVIGATE_TO

    override fun handle(message: BridgeMessage): String? {
        val payload = message.payload as? Map<*, *> ?: return null
        val url = payload["url"] as? String ?: return null
        // In a real app, this would navigate between Activities/fragments
        // For the scaffold, log the navigation
        android.util.Log.d("DuallerNav", "Navigate to: $url")
        return "{}"
    }
}
